src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,11 +3108,215 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
+            }
+
+            return initializedInstance;
+        }
+
+        private void HandleObjectOrCollectionInitializer(IObjectOrCollectionInitializerOperation initializer, IOperation initializedInstance)
+        {
+            // PROTOTYPE(dataflow): Handle collection initializers
+            IOperation previousInitializedInstance = _currentInitializedInstance;
+            _currentInitializedInstance = initializedInstance;
+
+            foreach (IOperation innerInitializer in initializer.Initializers)
+            {
+                handleInitializer(innerInitializer);
+            }
+
+            _currentInitializedInstance = previousInitializedInstance;
+
+            void handleInitializer(IOperation innerInitializer)
+            {
+                switch (innerInitializer.Kind)
+                {
+                    case OperationKind.MemberInitializer:
+                        handleMemberInitializer((IMemberInitializerOperation)innerInitializer);
+                        return;
+
+                    case OperationKind.SimpleAssignment:
+                        AddStatement(handleSimpleAssignment((ISimpleAssignmentOperation)innerInitializer));
+                        return;
+
+                    case OperationKind.CollectionElementInitializer:
+                        // PROTOTYPE(dataflow): support collection initializers
+                        throw new NotImplementedException();
+
+                    case OperationKind.Invalid:
+                        AddStatement(Visit(innerInitializer));
+                        return;
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(innerInitializer.Kind);
+                }
+            }
+
+            IOperation handleSimpleAssignment(ISimpleAssignmentOperation assignmentOperation)
+            {
+                // We spill all arguments to the lhs under the assumption that any of these could change after reading.
+                IOperation left = assignmentOperation.Target;
+                IOperation right;
+
+                switch (left.Kind)
+                {
+                    case OperationKind.FieldReference:
+                    case OperationKind.EventReference:
+                    case OperationKind.PropertyReference:
+                        var memberReference = (IMemberReferenceOperation)left;
+                        IPropertyReferenceOperation propertyReference = null;
+                        ImmutableArray<IArgumentOperation> propertyArguments = ImmutableArray<IArgumentOperation>.Empty;
+
+                        if (memberReference.Kind == OperationKind.PropertyReference &&
+                            !(propertyReference = (IPropertyReferenceOperation)memberReference).Arguments.IsEmpty)
+                        {
+                            var propertyArgumentsBuilder = ArrayBuilder<IArgumentOperation>.GetInstance();
```

Comment:
>var propertyArgumentsBuilder = ArrayBuilder<IArgumentOperation>.GetInstance(); [](start = 28, length = 78)

Is this a code path for initializer through an indexer?

Reply:
It will be.

---
In reply to: [185664894](https://github.com/dotnet/roslyn/pull/26389#discussion_r185664894) [](ancestors = 185664894)